host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -449,6 +449,39 @@ static void default_status(RASPIVID_STATE *state)
    raspicamcontrol_set_defaults(&state->camera_parameters);
 }
 
+static void check_camera_model(int cam_num)
+{
+   MMAL_COMPONENT_T *camera_info;
+   MMAL_STATUS_T status;
+
+   // Try to get the camera name
+   status = mmal_component_create(MMAL_COMPONENT_DEFAULT_CAMERA_INFO, &camera_info);
+   if (status == MMAL_SUCCESS)
+   {
+      MMAL_PARAMETER_CAMERA_INFO_T param;
+      param.hdr.id = MMAL_PARAMETER_CAMERA_INFO;
+      param.hdr.size = sizeof(param)-4;  // Deliberately undersize to check firmware version
+      status = mmal_port_parameter_get(camera_info->control, &param.hdr);
+
+      if (status != MMAL_SUCCESS)
+      {
+         // Running on newer firmware
+         param.hdr.size = sizeof(param);
+         status = mmal_port_parameter_get(camera_info->control, &param.hdr);
+         if (status == MMAL_SUCCESS && param.num_cameras > cam_num)
+         {
+            if (!strncmp(param.cameras[cam_num].camera_name, "toshh2c", 7))
+            {
+               fprintf(stderr, "The drivers for the TC358743 HDMI to CSI2 chip you are using are NOT supported.\n");
+               fprintf(stderr, "They were written for a demo, and are in the firmware on an as-is basis.\n");
+               fprintf(stderr, "Please don't ask for support or changes to them as they will not be changed.\n\n");
+            }
```

Comment:
This implies there is more than one driver - uses plural. Is that the case?
Perhaps combine second and third sentences 
"They were written for a demo purposes only, and are in the firmware on an as-is basis and therefor requests for support or changes will not be acted on." or similar.

Reply:
One could say that you have a driver for the chip and for the tuner, but fair enough to just call it one driver.
Reworded to yours (with spelling correction), and about to repush.